styx-api-service/src/main/java/com/spotify/styx/api/BackfillResource.java

```diff
@@ -417,4 +428,24 @@ private static boolean isActiveState(RunStateData runStateData) {
         : Stream.concat(processedStates.stream(), waitingStates.stream()).collect(toList());
   }
 
+  private RunStateData getRunStateData(Backfill backfill,
+      Map<WorkflowInstance, RunState> activeWorkflowInstances, Instant instant) {
+    final WorkflowInstance wfi = WorkflowInstance
+        .create(backfill.workflowId(), toParameter(backfill.schedule(), instant));
+    if (activeWorkflowInstances.containsKey(wfi)) {
+      RunState state = activeWorkflowInstances.get(wfi);
```

Comment:
The active state here might not be triggered by the backfill.

It might for example be a naturally triggered active state that is still running and that will be be re-triggered later by the backfill.

It could also be an instance that was ad-hoc re-triggered _after_ the backfill completed running. It could also be a an instance triggered by another backfill.

So for example, for a backfill that has already finished, we might return statuses that indicates that the backfill hasn't finished running yet. That seems incorrect.

Reply:
We seem to correctly check for the backfill trigger id in the replay code: https://github.com/spotify/styx/pull/546/files#diff-6762e6046164a0a7533c9d5d13973226R83

Reply:
This map should only contain states belonging to this backfill. Or?

Reply:
Indeed. Please add a test to verify this behavior. I.e. store an active state that is not triggered by the backfill we're querying for and verify that it is not used and that the events are replayed instead.

Reply:
Ok I think I understand.

Reply:
Addressed in later commit.